File: 2015/python/day2.py
with open('../input/day2.txt') as file:
    dimensionList = [line.strip() for line in file]

sqft = 0
ribbon = 0
for dimension in dimensionList:
    l = int(dimension.split('x')[0])
    w = int(dimension.split('x')[1])
    h = int(dimension.split('x')[2])

    sqft += 2*l*w + 2*w*h + 2*h*l
    sqft += min(l*w, w*h, h*l)

    ribbon += 2*l + 2*w + 2*h
    ribbon -= max(2*l, 2*w, 2*h)
    ribbon += l*w*h


print('partOne = ' + str(sqft))
print('partTwo = ' + str(ribbon))

# $ python3 day2.py
# partOne = 1588178
# partTwo = 3783758

File: 2015/rust/day2.rs
use std::fs;
use std::cmp;

fn main() {
    let contents = fs::read_to_string("../input/day2.txt").unwrap();
    let mut dims: [i32; 3] = [0; 3];
    let mut surface_area = 0;
    let mut scrap = 0;
    let mut ribbon = 0;

    for gift in contents.lines() {
        for (i, elem) in gift.split('x').enumerate() {
            dims[i] = elem.parse().unwrap();
        }

        surface_area += (2 * dims[0] * dims[1]) + (2 * dims[1] * dims[2]) + (2 * dims[0] * dims[2]);
        scrap += cmp::min(cmp::min(dims[0] * dims[1], dims[0] * dims[2]), dims[1] * dims[2]);

        ribbon += dims[0]*2 + dims[1]*2 + dims[2]*2;
        ribbon -= cmp::max(cmp::max(dims[0]*2, dims[1]*2), dims[2]*2);
        ribbon += dims[0] * dims[1] * dims[2];
    }

    println!("partOne = {}", surface_area + scrap);
    println!("partTwo = {}", ribbon);
}